spec/unit/type/file/content_spec.rb

```diff
@@ -184,25 +184,30 @@
           expect(content.respond_to?("diff")).to eq(false)
         end
 
-        [true, false].product([true, false]).each do |cfg, param|
```

Comment:
It looks like you lost some negative tests on the overall Puppet[:show_diff] / Resource[show_diff] matrix here?